Anytype/Sources/PresentationLayer/Templates/Views/Selection/Objects/TemplatePreview.swift

```diff
@@ -0,0 +1,114 @@
+import SwiftUI
+import Services
+
+struct TemplatePreview: View {
+    let viewModel: TemplatePreviewModel
+    
+    var body: some View {
+        ZStack {
+            Rectangle()
+                .fill(.clear)
+            content
+        }
+        .border(
+            16,
+            color: viewModel.isDefault ?
+            Color.System.amber50 : .Stroke.primary,
+            lineWidth: viewModel.isDefault ? 2 : 1
+        )
+        .cornerRadius(16, corners: .top)
+        .frame(width: 120, height: 224)
+    }
+    
+    var content: some View {
+        VStack(alignment: viewModel.alignment.horizontalAlignment) {
+            cover
+            switch viewModel.model {
+            case .blank:
+                AnytypeText(
+                    "Blank",
+                    style: .caption2Medium,
+                    color: .Text.tertiary
+                ).padding(.horizontal, 16)
+            case let .installed(templateModel):
+                AnytypeText(
+                    templateModel.title,
+                    style: .caption2Medium,
+                    color: .Text.primary
+                ).padding(.horizontal, 16)
+            }
+            Spacer.fixedHeight(12)
+            shimmer.padding(.horizontal, 16)
+            Spacer()
+        }
+    }
+    
+    private var cover: some View {
+        Group {
+            if case let .installed(templateModel) = viewModel.model {
+                switch templateModel.header {
+                case .filled(let state, _):
+                    ObjectHeaderFilledContentSwitfUIView(
+                        configuration: ObjectHeaderFilledConfiguration(
+                            state: state,
+                            isShimmering: false,
+                            sizeConfiguration: .templatePreviewSizeConfiguration
+                        )
+                    )
+                default:
+                    Spacer.fixedHeight(28)
+                }
+            } else {
+                Spacer.fixedHeight(28)
+            }
+        }
+    }
+    
+    var shimmer: some View {
```

Comment:
Why you don't use `.redacted(.placeholder)` for shimmering state?

Reply:
It is a custom shimmer. It should not shimmer some view's state, just view some placeholders without content 
<img width="377" alt="Screenshot 2023-07-19 at 17 28 42" src="https://github.com/anyproto/anytype-swift/assets/15055998/97893e32-abd0-4985-8860-8c67c6f04b6a">